src/components/Vesting/RewardLockerSchedules.tsx

```diff
@@ -0,0 +1,195 @@
+  const above768 = useMedia('(min-width: 768px)')
+  const [expanded, setExpanded] = useState<boolean>(true)
+  const { vestMultipleTokensAtIndices } = useVesting(rewardLockerAddress)
+
+  if (!schedules) {
+    schedules = []
+  }
+
+  const info = schedules.reduce<{
+    [key: string]: {
+      vestableIndexes: number[]
+      vestableAmount: BigNumber
+      fullyIndexes: number[]
+      fullyAmount: BigNumber
+      totalAmount: BigNumber
+      unlockedAmount: BigNumber
+      token: Token
+    }
+  }>((result, schedule) => {
+    if (!currentBlockNumber) return result
+    const address = (schedule[4] as Token).symbol as string
+    if (!result[address]) {
+      result[address] = {
+        vestableIndexes: [],
+        vestableAmount: BigNumber.from(0),
+        fullyIndexes: [],
+        fullyAmount: BigNumber.from(0),
+        totalAmount: BigNumber.from(0),
+        unlockedAmount: BigNumber.from(0),
+        token: schedule[4] as Token
+      }
+    }
+
+    result[address].totalAmount = result[address].totalAmount.add(BigNumber.from(schedule[2]))
+    /**
+     * fullyVestedAlready = schedule.quantity - schedule.vestedQuantity
+     */
+    const fullyVestedAlready = BigNumber.from(schedule[2])
+      .sub(BigNumber.from(schedule[3]))
+      .isZero()
+
+    /**
+     * isEnd = schedule.endBlock - currentBlock >= 0
+     */
+    const isEnd = !BigNumber.from(currentBlockNumber)
+      .sub(BigNumber.from(schedule[1]))
+      .isNegative()
+    // const vestedAndVestablePercent = BigNumber.from(currentBlockNumber)
+    //   .sub(BigNumber.from(s[1]))
+    //   .isNegative()
+    //   ? BigNumber.from(currentBlockNumber)
+    //       .sub(BigNumber.from(s[0]))
+    //       .mul(100)
+    //       .div(BigNumber.from(s[1]).sub(BigNumber.from(s[0])))
+    //   : 100
+    // const unlockedAmount = BigNumber.from(s[2])
+    //   .mul(vestedAndVestablePercent)
+    //   .div(100)
+    const unlockedAmount = isEnd
+      ? BigNumber.from(schedule[2])
+      : BigNumber.from(schedule[2])
+          .mul(BigNumber.from(currentBlockNumber).sub(BigNumber.from(schedule[0])))
+          .div(BigNumber.from(schedule[1]).sub(BigNumber.from(schedule[0])))
+    const vestableAmount = unlockedAmount.sub(BigNumber.from(schedule[3])) // vestableAmount = unlock - vestedQuanitty
+    if (!fullyVestedAlready) {
+      result[address].vestableIndexes.push(schedule[5])
+    }
+    result[address].vestableAmount = result[address].vestableAmount.add(
+      vestableAmount.isNegative() ? BigNumber.from(0) : vestableAmount
+    )
+
+    if (!fullyVestedAlready && !!currentBlockNumber && currentBlockNumber > schedule[1]) {
+      result[address].fullyIndexes.push(schedule[5])
+      result[address].fullyAmount = result[address].fullyAmount.add(BigNumber.from(schedule[2]))
+    }
+
+    result[address].unlockedAmount = result[address].unlockedAmount.add(unlockedAmount)
+    return result
+  }, {})
+
+  const [, setPendingTx] = useState(false)
```

Comment:
do we need this state?

Reply:
I just reused Tien's code.
But now I realized that we can show a TransactionConfirmationModal here, it's much better for UX
Thanks

Reply:
I updated this to use a TransactionConfirmationModal